pkg/systemd/services/services.jsx

```diff
@@ -216,7 +216,7 @@ class ServicesPageBody extends React.Component {
     }
 
     onOptionsChanged(options) {
-        const currentOptions = { ...cockpit.location.options, ...options };
+        const currentOptions = { ...this.props.options, ...options };
```

Comment:
Then I think we should go all the way and *always* use cockpit.location.options, and stop propagating "options" as a property. IMHO there should only be one source of truth, not two -- otherwise it's prone to race conditions again. Then we should bring back the `locationchanged` handler, and drop componentDidUpdate(), as we wouldn't get a property change any more. WDYT?